tests/workers/test_cvecheck.py

```diff
@@ -119,8 +114,84 @@ def test_python_mako(self):
         assert detail['id'] == 'CVE-2010-2480'
         assert 'http://bugs.python.org/issue9061' in detail['references']
         assert detail['severity'] == 'Medium'
+
+    def test_python_requests(self):
+        """ To make sure that python CPE suppression works (issue#131) """
+        extracted = os.path.join(
+                        os.path.dirname(
+                         os.path.abspath(__file__)), '..', 'data', 'pypi', 'requests-2.5.3')
+        args = {'ecosystem': 'pypi', 'name': 'requests', 'version': '2.5.3'}
+        flexmock(EPVCache).should_receive('get_extracted_source_tarball').and_return(extracted)
+        task = CVEcheckerTask.create_test_instance(task_name='source_licenses')
+        results = task.execute(arguments=args)
+
+        assert isinstance(results, dict)
+        assert set(results.keys()) == {'details', 'status', 'summary'}
         assert results['status'] == 'success'
-        assert results['summary'] == ['CVE-2010-2480']
+        assert results['summary'] == ['CVE-2015-2296']
+        details = results.get('details')
+        # http://www.cvedetails.com/version/180634/Python-requests-Requests-2.5.3.html
+        #  "details": [
+        #    {
+        #        "cvss": {
+        #            "score": 6.8,
+        #            "vector": "AV:N/AC:M/Au:?/C:P/I:P/A:P"
+        #        },
+        #        "description": "The resolve_redirects function in sessions.py ...",
+        #        "id": "CVE-2015-2296",
+        #        "references": [
+        #            "http://advisories.mageia.org/MGASA-2015-0120.html"
+        #        ],
+        #        "severity": "Medium"}]
+        assert isinstance(details, list) and len(details) == 1
+        detail = details[0]
+        assert set(detail.keys()) == {'cvss', 'description', 'id', 'references', 'severity'}
+        cvss = detail['cvss']
+        assert cvss['score'] == 6.8
+        assert cvss['vector'] == 'AV:N/AC:M/Au:?/C:P/I:P/A:P'
+        assert detail['id'] == 'CVE-2015-2296'
+        assert 'http://advisories.mageia.org/MGASA-2015-0120.html' in detail['references']
+        assert detail['severity'] == 'Medium'
```

Comment:
Something like [datadiff](https://pypi.python.org/pypi/datadiff) could make this (and the test bellow) more readable and less error prone.